### PR TITLE
Update to latest version of atat

### PR DESCRIPTION
**Pull request by fkohlgrueber**
This PR updates `espresso` to work with the current version of `atat`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,12 +19,9 @@ numtoa = "0.2"
 [dev-dependencies]
 env_logger = "0.6"
 log = "0.4"
-serialport = { git = "https://gitlab.com/dbrgn/serialport-rs", branch = "embedded", features = ["embedded"], default_features = false }
+serialport = { git = "https://gitlab.com/fkohlgrueber/serialport-rs", branch = "embedded", features = ["embedded"], default_features = false }
```
Comment (fkohlgrueber) [inline Cargo.toml:22]: This change is temporary and can be reverted as soon as https://gitlab.com/dbrgn/serialport-rs/-/merge_requests/1 is merged.

Comment (dbrgn) [reply Cargo.toml:22]: Thanks! That reminded me of https://gitlab.com/susurrus/serialport-rs/-/merge_requests/76, which I just resurrected.

Comment (fkohlgrueber) [reply Cargo.toml:22]: Seems like serialport-rs is seeking a new maintainer (https://gitlab.com/susurrus/serialport-rs/-/issues/111), so I'm not sure if / when your PR will get merged.

**Comment by fkohlgrueber** [comment]
How do you feel about this PR? Atat has in the meantime updated to const generics, but I think it's anyway a good idea to merge this PR before updating to that new Atat version.

**Comment by fkohlgrueber** [comment]
Update done, can you take a look? Thanks in advance!

**Comment by dbrgn** [comment]
I already started a review, but didn't finish it yet, sorry. No time this weekend so far.

I'll try to do it tomorrow.

File: Cargo.toml (hunk 1 above)
Comment (dbrgn) [reply Cargo.toml:24]: True. Let's keep your branch for now, we can still decide what to do once embedded-hal 1.0 is released.

File: Cargo.toml [hunk 2]
```diff
@@ -9,22 +9,21 @@ readme = "README.md"
 edition = "2018"
 
 [dependencies]
-atat = { git = "https://github.com/BlackbirdHQ/atat", branch = "master" }
-embedded-hal = "0.2"
-heapless = "0.5"
+#atat = { git = "https://github.com/BlackbirdHQ/atat", branch = "master" }
+# atat = { path = "../atat/atat" }
```
Comment (dbrgn) [inline Cargo.toml:13]: Can you remove this line? (Feel free to leave in the commented-out git source on line 12 if you think that'll become useful.)

**Review by dbrgn — CHANGES_REQUESTED** [review]
Cool, thanks for your work! Left a few comments.

**Comment by dbrgn** [comment]
@fkohlgrueber do you plan to continue working on this?

File: src/commands/requests.rs [hunk 3]
```diff
@@ -20,12 +22,17 @@ pub struct At;
...
+        Vec::from_slice(b"AT\r\n").unwrap()
     }
 
-    fn parse(&self, resp: &str) -> Result<Self::Response, atat::Error> {
+    fn parse(
+        &self,
+        resp: Result<&[u8], &InternalError>,
+    ) -> Result<Self::Response, Error<Self::Error>> {
+        let resp = core::str::from_utf8(resp?).unwrap();
         if !resp.trim().is_empty() {
```
[the remark this answers is not in the document]
Comment (fkohlgrueber) [reply src/commands/requests.rs:35]: Let's do this in a separate issue. This PR has been open for too long already.

**Comment by fkohlgrueber** [comment]
Hi @dbrgn,
I'm sorry for taking so long to follow up on this. I'd like to use espresso for a hobby project at some point in the future, but this doesn't have any priority and I don't have much time to contribute anymore. I should've responded earlier anyway, sorry for that. 
I've fixed the issues you pointed out in your review. Is this ready to be merged now?

**Comment by dbrgn** [comment]
Oops, seems that compilation doesn't currently work. Seems that the reason is yet another embedded-hal update.

I'll lock it to alpha 4 for now, then we can fix this later.

**Pull request merged**